### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by rbanos-mv**
# Vet clinic

> In this project, I will use a relational database to create the initial data structure for a vet clinic. I will create a table to store animals' information, insert some data into it, and query it.

# This PR includes the following requirements:

### Project requirements
- Add a column `species` of type `string` to your `animals` table. Modify your `schema.sql` file.
- Insert the following data:
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Remember to add these insertions to your `data.sql` file.

----
## NOTE: When neuterd is FALSE, it shows an empty cell
image

- Inside a transaction update the `animals` table by setting the `species` column to `unspecified`. Verify that change was made. Then roll back the change and verify that the `species` columns went back to the state before the transaction.
image
image
image

- Inside a transaction:
  - Update the `animals` table by setting the `species` column to `digimon` for all animals that have a name ending in `mon`.
  - Update the `animals` table by setting the `species` column to `pokemon` for all animals that don't have `species` already set.
  - Commit the transaction.
  - Verify that change was made and persists after commit.
 image
image
- **Take a screenshot of the results of your actions.**
- Now, take a deep breath and... **Inside a transaction** delete all records in the `animals` table, then roll back the transaction.
- After the rollback verify if all records in the `animals` table still exists. After that, you can start breathing as usual ;) 
image
- **Take a screenshot of the results of your actions.**
- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
image
image
- **Take a screenshot of the results of your actions.**

----
- Write queries to answer the following questions:
  - How many animals are there?
image
  - How many animals have never tried to escape?
image
  - What is the average weight of animals?
image
  - Who escapes the most, neutered or not neutered animals?
image
  - What is the minimum and maximum weight of each type of animal?
image
  - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
image

- **Take a screenshot of the results of your queries.**
- Remember to add these queries to your `queries.sql` file.
- Remember to add screenshots with the results of your queries and with the results of your play with transactions to your Pull Request description.

### General requirements

- Make sure that you used the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Project requirements
- Use [this template](https://github.com/microverseinc/curriculum-template-databases) to generate your repo.
- Create a database named `vet_clinic`.
- Create a table named `animals` with the following columns:
  - `id`: integer
  - `name`: string
  - `date_of_birth`: date
  - `escape_attempts`: integer
  - `neutered`: boolean
  - `weight_kg`: decimal
- Copy the SQL you used in the previous point into the `schema.sql` file in the generated repository.
- Insert the following data:
  - Animal: His name is Agumon. He was born on Feb 3rd, 2020, and currently weighs 10.23kg. He was neutered and he has never tried to escape.
  - Animal: Her name is Gabumon. She was born on Nov 15th, 2018, and currently weighs 8kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Pikachu. He was born on Jan 7th, 2021, and currently weighs 15.04kg. He was not neutered and he has tried to escape once.
  - Animal: Her name is Devimon. She was born on May 12th, 2017, and currently weighs 11kg. She is neutered and she has tried to escape 5 times.
- Copy the SQL you used in the previous point into the `data.sql` file in the generated repository.
- Write queries for the following quests:
  - Find all animals whose name ends in "mon".
image

  - List the name of all animals born between 2016 and 2019.
image

  - List the name of all animals that are neutered and have less than 3 escape attempts.
image

  - List the date of birth of all animals named either "Agumon" or "Pikachu".
image

  - List name and escape attempts of animals that weigh more than 10.5kg
image

  - Find all animals that are neutered.
image

  - Find all animals not named Gabumon.
image

  - Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
image

- **Take a screenshot of the results of your queries.**
- Copy the SQL you used in the previous point into the `queries.sql` file in the generated repository.
- Remember to add the screenshot with the results of your queries to your Pull Request description.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -7,3 +7,12 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
 (4, 'Devimon', '2017-05-12', 5, true, 11.00)
 ;
 
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES 
+(5, 'Charmander', '2020-02-08', 0, false, -11),
+(6, 'Plantmon', '2021-11-15', 2, true, -5.7),
```
Comment (arslanbisharat) [inline data.sql:12]: - [ ] date_of_birth for `Plantmon` is 2022/11/15.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (rbanos-mv) [reply data.sql:12]: My dear friend... My eyes are not good enough, so I cannot see the **2022**  :smile: 
image

Comment (rbanos-mv) [reply data.sql:12]: The previous reviewer (@arslanbisharat) is asking to change the date of birth for **Plantmon** to **2022**, but according to the requirement's document, it should be **2021**. Following is the screenshot from the requirement's document
image

**Review by keneogu — APPROVED** [review]
Hi @rbanos-mv,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

## Optional suggestions

- [ ] The calculator design is too small, you can make it large.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (arslanbisharat) [reply data.sql:12]: I saw this issue after your project review. There are some confusions between our guidelines and yours so I have reported that to team

**Pull request merged**